cpu-performance.rst: add thermal configuration section (#961)
clearlinux/clear-linux-documentation: source/guides/maintenance/cpu-performance.rst

   After adding a new group, you must log out and log back in for the new group
   to take effect.

> Thermal configuration

Reviewer: Maybe a better title for this section is "Enhanced thermal configuration"

=====================

> Better thermal control and performance can be archived by providing platform

Reviewer 2: ```suggestion
Better thermal control and performance can be achieved by providing platform
```

> specified configuration to thermald.

Reviewer: ```suggestion
specific configuration to thermald.
```

`Linux DPTF Extract Utility`_ is a companion tool to thermald, This tool can

> make use of "Intel ® Dynamic Platform and Thermal Framework (`Intel DPTF`_)"

Reviewer 2: ```suggestion
make use of :abbr:`Intel® Dynamic Platform and Thermal Framework (`Intel DPTF)`
```

Reviewer: I see this was not packaged natively in Clear Linux because it is closed-source: https://github.com/clearlinux/distribution/issues/1073 

We should mention why it is not bundled and the limitations.

technology, and convert to the thermal_conf.xml configuration format used
by thermald.

> First, make sure thermald version is equal or above 1.9.1:

Reviewer 2: ```suggestion
First, make sure thermald version is equal to or above version 1.9.1:
```

Reviewer: I'd rather make sure the version in Clear Linux is at least this version before publishing to avoid cluttering with static version check information. That sort of thing tends to be irrelevant quickly with the pace that Clear Linux updates move.

Author: Current version in Clear Linux is 1.9.
I submit a patch to upgrade to 1.9.1, and is under review now.

.. code:: bash

   thermald --version

> Otherwise, you need to add an option :command:`--ignore-default-control` to

Reviewer 2: ```suggestion
If its version is not updated, append the  :command:`--ignore-default-control` to ``ExecStart``,  as shown below.
```

thermald:

.. code:: bash

   # edit thermald.servcie

>    vim /usr/lib/systemd/system/thermald.service

Reviewer 2: ```suggestion
sudo  vim /usr/lib/systemd/system/thermald.service
```

Reviewer: See comment above about avoiding this whole section, but in Clear Linux editing file under `/usr/` is not safe and prone to being overwritten by system update. See the documentation on stateless. 

If you have to do this in the future on Clear Linux system, a better approach is to create a modified service file in `/etc/`. The `systemctl edit --full thermald.service` can help do this neatly. Even better would be a drop-in directory for the unit. https://www.freedesktop.org/software/systemd/man/systemd.unit.html

>    # append --ignore-default-control option to ExecStart line, like below

Reviewer 2: Remove Ln 220 (it now appears above).

>    # ExecStart=/usr/bin/thermald --no-daemon --dbus-enable --ignore-default-control

Reviewer 2: Make Ln 221 a separate example, like so: 
```suggestion
.. code:: bash
   ExecStart=/usr/bin/thermald --no-daemon --dbus-enable --ignore-default-control
```
Assure that this line is **NOT** commented out with a '#'

   # reload

>    sudo systemctl daemon-reload

Reviewer 2: ```suggestion
 .. code-block:: bash
    sudo systemctl daemon-reload
```

Then, generate thermal configuration as below:

.. code:: bash

   git clone https://github.com/intel/dptfxtract.git
   cd dptfxtract
   sudo acpidump > acpi.out
   acpixtract -a acpi.out

>    sudo ./dptfxtract *.dat

Reviewer 2: When I tested these commands, `sudo ./dptfxtract *.dat` produced the following: 
```
...
No valid tables found
```
Is this an error?

Author: DPTF need BIOS support. Most Intel laptop support this feature.
Typical path is: Configuration -> DPTF (Enabled)

Author: I can not find such a guide/article to enable DPTF in BIOS. And menu entry might changed from different vendor.

And from the project of dptf for chromimum OS, here is what they says:

> Requirement: DPTF requires corresponding ACPI support in BIOS. Not all
Intel based platforms support DPTF in BIOS. Please contact your BIOS vendor
to see if DPTF is enabled in your system.

https://github.com/intel/dptf

Reviewer 2: Given that the option to enable may vary per the BIOS, I'll let this stand as is. Refinements to this will be addressed in a follow-up GH Issue.

thermald configuration files will be generated and saved to
:command:`/etc/thermal/` folder. Restart thermald service to take effect.

.. code:: bash

   sudo systemctl restart thermald.service

> 

Reviewer 2: After running this command, I ran again the command ``thermald --version``. The output was ```1.9``, not ``1.9.1``. 

Should ``thermald`` have been upgraded? Or, was this prevented due to the error on "tables" from above?   
Can you include a command that will confirm that the above changes took effect? In other words: What change should be noticeable from the standpoint of a user?

Author: No, modifying `thermald.service` will not change `thermald` version, it's a workaround for version under `1.9`.

User can check `thermald` status by `systemctl status thermald.service`, and find log like:
```
Dec 06 05:44:54 icl-clr-debug-i5 thermald[459]: [WARN]Using generated /etc/thermald/thermal-conf.xml.auto
```
it indicates that the configure file has been used by `thermald`.

Reviewer 2: That's fair.  Please include a statement to that effect. Something along the lines of "This is a workaround..."

Author: I just remove this workaround part, and suggest user to upgrade to latest Clear Linux version.

Reviewer 2: OK. Please add the command to update, shown on Line 219 above.

   After adding a new group, you must log out and log back in for the new group
   to take effect.

Enhanced thermal configuration
===============================

Better thermal control and performance can be achieved by providing platform
specific configuration to thermald.

`Linux DPTF Extract Utility`_ is a companion tool to thermald, This tool can
make use of :abbr:`Intel® Dynamic Platform and Thermal Framework (`Intel DPTF)`
technology, and convert to the thermal_conf.xml configuration format used
by thermald. It's a closed-source project, and unable to be packaged as bundle
in Clear Linux OS, so we need to follow below steps to generate configuration.

Intel DPTF requires BIOS support, it's typically used by laptop.
The first step is to make sure your machine's BIOS has DPTF feature
and is enabled.

Then, check thermald version is equal to or above version 1.9.1:

.. code:: bash

   thermald --version

If its version is not up-to-date, please upgrade to latest Clear Linux version.

Next step, generate thermal configuration as below:

.. code:: bash

   git clone https://github.com/intel/dptfxtract.git
   cd dptfxtract

>    sudo acpidump > acpi.out

Reviewer: Can you include a step to ensure the bundle that contains these tools (acpica-unix2) is installed (i.e. `sudo swupd bundle-add acpica-unix2`

Author: fixed in [0abdc54](https://github.com/clearlinux/clear-linux-documentation/pull/961/commits/0abdc549c8f9f41c68bca48e6b623365ea6604d2)

   After adding a new group, you must log out and log back in for the new group
   to take effect.

Enhanced thermal configuration
===============================

Better thermal control and performance can be achieved by providing platform
specific configuration to thermald.

`Linux DPTF Extract Utility`_ is a companion tool to thermald, This tool can
make use of :abbr:`Intel® Dynamic Platform and Thermal Framework (`Intel DPTF)`
technology, and convert to the thermal_conf.xml configuration format used
by thermald. It's a closed-source project, and unable to be packaged as bundle
in Clear Linux OS, so we need to follow below steps to generate configuration.

Intel DPTF requires BIOS support, it's typically used by laptop.
The first step is to make sure your machine's BIOS has DPTF feature
and is enabled.

Then, check thermald version is equal to or above version 1.9.1:

.. code:: bash

   thermald --version

If its version is not up-to-date, please upgrade to latest Clear Linux version.

> 

Reviewer 2: ```suggestion
.. code-block:: bash 

   sudo swupd update

```

Next step, generate thermal configuration as below:

.. code:: bash

   sudo swupd bundle-add acpica-unix2  # install acpi tools
   git clone https://github.com/intel/dptfxtract.git
   cd dptfxtract
   sudo acpidump > acpi.out
   acpixtract -a acpi.out
   sudo ./dptfxtract *.dat

thermald configuration files will be generated and saved to
:command:`/etc/thermal/` folder. Restart thermald service to take effect.

.. code:: bash

   sudo systemctl restart thermald.service

check whether the configuration is in used.

.. code:: bash

   sudo systemctl status thermald.service

> 

Reviewer 2: Please state what message (stdout) the user should see if configuration is in use. (This can be console output too.)

Author: already mentioned in L244.
